runtime/gc_base/StringTable.cpp

```diff
@@ -533,9 +533,16 @@ j9gc_createJavaLangString(J9VMThread *vmThread, U_8 *data, UDATA length, UDATA s
 	j9object_t result = NULL;
 	j9object_t charArray = NULL;
 	UDATA allocateFlags = J9_GC_ALLOCATE_OBJECT_NON_INSTRUMENTABLE;
+	bool isUnicode = J9_ARE_ANY_BITS_SET(stringFlags, J9_STR_UNICODE);
 
-	if (VM_VMHelpers::isUTF8ASCII(data, length)) {
-		stringFlags |= J9_STR_ASCII;
+	if (isUnicode && IS_STRING_COMPRESSION_ENABLED_VM(vm)) {
+		if (VM_VMHelpers::isUnicodeASCII((const U_16 *)data, length / 2)) {
+			stringFlags |= J9_STR_ASCII;
+		}
+	} else {
+		if (VM_VMHelpers::isUTF8ASCII(data, length)) {
```

Comment:
This is still possibly called when isUnicode is true, due to the compound condition in the if above.

Reply:
Oops. Fixed.